tests/unit/test_hardware.py

```diff
@@ -2,12 +2,108 @@
 import unittest
 from unittest import mock
 
-from hardware import get_bmc_address, lshw
+import pytest
+
+from hardware import _split_at, get_bmc_address, hwinfo, lshw
+
+
+@pytest.mark.parametrize(
+    "s,delimiter,num,expect",
+    [("abc-d1-d1-d1-def", "d1", 2, ("abc-d1-", "-d1-def"))],
+)
+def test_split_at(s, delimiter, num, expect):
+    output = _split_at(s, delimiter, num)
+    case = unittest.TestCase()
+    case.assertCountEqual(output, expect)
+
+
+class TestHwinfo:
```

Comment:
```suggestion
class TestHwinfo(unittest.TestCase):
```
Should inherit from `unittest.TestCase` to include as part of tests.

Reply:
Got to know we're using pytest to manage this tests. The other test classes would need to be migrated over too.